### PR TITLE
Math magicians: fetch data from API

### PR DESCRIPTION
**Pull request by rudicarrilloypr**
### Project requirements

- [x] - Use the following external API: [API Ninjas](https://api-ninjas.com/api/quotes)
- [x] - Create a new component for displaying a quote
- [x] - Fetch data inside of a new component
- [x] - Show a `loading` state
- [x] - Show an `error` state
- [x] - Import the new component into your main component (next to the Calculator)
- [x]   - It should be on the same page as the Calculator

- [x] _Note: Practice what you have learned until now: use `useEffect` and `props`._

### CODE REVIEW

File: src/components/Quote.jsx [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+        if (response.ok) return response.json();
+        throw response;
+      })
+      .then((data) => {
+        setQuoteData(data[0]); // We only need the first quote
+        setLoading(false);
+      })
+      .catch((error) => {
+        // eslint-disable-next-line no-console
+        console.error('Error fetching data: ', error);
```
Comment (AdnanOlarmmi) [inline src/components/Quote.jsx]: - [x] Kindly remove unnecessary `console.log` statements from your code 😉, as statement like this litters your code.

File: src/components/Quote.jsx [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+      .then((response) => {
+        if (response.ok) return response.json();
+        throw response;
+      })
+      .then((data) => {
+        setQuoteData(data[0]); // We only need the first quote
+        setLoading(false);
+      })
+      .catch((error) => {
+        // eslint-disable-next-line no-console
```
Comment (AdnanOlarmmi) [inline src/components/Quote.jsx]: - [x] Kindly ensure not to disable linters in any part of your code/project as linters is what ensures that proper industry standard is followed. 😉

File: src/components/Quote.jsx [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+        console.error('Error fetching data: ', error);
+        setError(error);
+      });
+  }, []); // Empty dependency array means this effect runs once on mount
```
Comment (AdnanOlarmmi) [inline src/components/Quote.jsx]: - [x] [OPTIONAL] Kindly note that comments are not supposed to duplicate codes, you can remove the statement, ` // Empty dependency array means this effect runs once on mount`. Check out [this](https://stackoverflow.blog/2021/12/23/best-practices-for-writing-code-comments/) article for more.

File: src/components/Quote.jsx [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+      headers: {
+        'X-Api-Key': 'oF/Wh13s4YEHx1jaPyWlmw==eYn2l3wmuDgNIs36',
+      },
+    })
+      .then((response) => {
+        if (response.ok) return response.json();
+        throw response;
+      })
+      .then((data) => {
+        setQuoteData(data[0]); // We only need the first quote
```
Comment (AdnanOlarmmi) [inline src/components/Quote.jsx]: - [x] [OPTIONAL] Kindly note that comments are not supposed to duplicate codes, you can remove the statement, ` // We only need the first quote`. Check out [this](https://stackoverflow.blog/2021/12/23/best-practices-for-writing-code-comments/) article for more.

**Review by AdnanOlarmmi — CHANGES_REQUESTED** [review]
Hi @rudicarrilloypr, 

This is Adnan Is-haq, your beloved code reviewer,

You've done a very good job so far, but a few things still need to be worked on before moving on to the next project, but you're so close!

You're Almost there, funny one eyed alien

## Highlights
- Effective use of the UseEffect hook.
- Error handled properly,

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
- N/A

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Reem-lab — APPROVED** [review]
# Hi @rudicarrilloypr,  👋🏻👋🏻

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 


## Highlights

- [x]  No Linters errors 🔥
- [x]  Great job working on the functionality💯
- [x]  Kudos for making the requested changes🔥
- [x]  Followed GitFlow ⭐🚀


## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**